### PR TITLE
build_config: support the latest release of pkg:yaml

### PR DESCRIPTION
**Pull request by kevmoo**
Pass Uri instead of String when parsing BuildConfig

### CODE REVIEW

File: build_config/pubspec.yaml [hunk 1]
```diff
@@ -21,6 +21,8 @@ dev_dependencies:
   test: ^1.6.0
 
 dependency_overrides:
+  # just for testing...
```
Comment (kevmoo) [inline build_config/pubspec.yaml:24]: Forcing 3.0.0 here – just to sanity check CI

File: build_config/lib/src/build_config.dart [hunk 2]
```diff
@@ -109,7 +109,7 @@ class BuildConfig {
         (map) =>
             BuildConfig.fromMap(packageName, dependencies, map ?? const {}),
         allowNull: true,
-        sourceUrl: configYamlPath,
+        sourceUrl: Uri.file(configYamlPath),
```
Comment (jakemac53) [inline build_config/lib/src/build_config.dart:112]: This can be null it looks like, at least in tests

Comment (kevmoo) [reply build_config/lib/src/build_config.dart:112]: yup...

Comment (kevmoo) [reply build_config/lib/src/build_config.dart]: fixed

**Comment by kevmoo** [comment]
I'm hitting analyzer version issues. I think we've proved that the fix is okay – just hitting bumps w/ the versions of packages

**Comment by kevmoo** [comment]
@jakemac53 – this should work now!

File: build_config/pubspec.yaml [hunk 3]
```diff
@@ -21,6 +21,8 @@ dev_dependencies:
...
+  yaml: 3.0.0-nullsafety.0
```
Comment (jakemac53) [inline build_config/pubspec.yaml:25]: I would go ahead and remove this now

Comment (kevmoo) [reply build_config/pubspec.yaml:25]: Yup! just making sure we were green first

**Review by natebosch — APPROVED** [review]

**Review by jakemac53 — APPROVED** [review]

File: build_config/pubspec.yaml (hunk 3 above)
Comment (kevmoo) [reply build_config/pubspec.yaml]: done

File: build_config/pubspec.yaml [hunk 4]
```diff
@@ -21,6 +21,8 @@ dev_dependencies:
...
   # Required due to the tight dependency on this package from these packages
   build_runner:
```
Comment (kevmoo) [inline build_config/pubspec.yaml]: @jakemac53 – should I remove these, too?

**Pull request merged**